R/obic_score.R

```diff
@@ -28,27 +65,24 @@ obic_score <- function(dt.ind) {
   w <- dcast(w,rsid~var,value.var = 'weight')
   
   # Score the chemical indicators
-  dt.ind[, S_C := 	w$W_C_N * I_C_N + w$W_C_P * I_C_P + w$W_C_K * I_C_K + 
-					w$W_C_MG * I_C_MG + w$W_C_S * I_C_S + w$W_C_PH * I_C_PH + 
-					w$W_C_CEC * I_C_CEC + w$W_C_CU * I_C_CU + w$W_C_ZN * I_C_ZN]
+  dt.ind[, S_C_A := ((w$W_C_N + 1/I_C_N) * I_C_N + (w$W_C_P + 1/I_C_P) * I_C_P + w$W_C_K * I_C_K + 
+					w$W_C_MG * I_C_MG + w$W_C_S * I_C_S +(w$W_C_PH + 1/I_C_PH) * I_C_PH + 
+					w$W_C_CEC * I_C_CEC + w$W_C_CUZN * (I_C_CU + I_C_ZN)) / 
+					  (1 + 1/I_C_N + 1/I_C_P + 1/I_C_PH)]
   
   # Score the physical indicators
-   dt.ind[, S_P :=  w$W_P_CR * I_P_CR + w$W_P_SE * I_P_SE + w$W_P_MS * I_P_MS +  
+   dt.ind[, S_P_A :=  w$W_P_CR * I_P_CR + w$W_P_SE * I_P_SE + w$W_P_MS * I_P_MS +  
 					w$W_P_BC * I_P_BC + w$W_P_DU * I_P_DU + w$W_P_CO * I_P_CO + 
 					w$W_P_CEC * I_P_CEC + w$W_P_WRI * I_P_WRI]
   
   # Score the biology
-  dt.ind[, S_B := w$W_B_DI * I_B_DI + w$W_B_SF * I_B_SF + w$W_B_SB * I_B_SB]
+  dt.ind[, S_B_A := w$W_B_DI * I_B_DI + w$W_B_SF * I_B_SF + w$W_B_SB * I_B_SB]
   
   # Score the management
-  dt.ind[, S_M := I_M]
+  dt.ind[, S_M_A := I_M]
   
   # Calculate the total score
-  dt.ind[, S_T := 0.3*S_C + 0.3*S_P + 0.3*S_B + 0.1*S_M]
-  
-  # Aggregate per field over the last 10 years
-  col.sel <- colnames(dt.ind)[grepl("ID|^I_|^S_", colnames(dt.ind))]
-  dt.ind <- dt.ind[, lapply(.SD, mean), by = ID, .SDcols = col.sel]
+  dt.ind[, S_T_A := 0.35 * S_C_A + 0.35 * S_P_A + 0.2 * S_B_A + 0.1 * S_M_A]
```

Comment:
how is the environmental performance included? or is that done in a separate branch?

Reply:
Environmental performance is not included in the score yet, due to the discussion if the definition 'sustainable agricultural production' includes environmental impact. Also because environmental performance contains only N leaching to surface water and groundwater it is not a complete proxy for the effects on the environment